plugins/uefi/fu-uefi-common.c

```diff
@@ -241,12 +244,17 @@ fu_uefi_get_esp_path_for_os (const gchar *base)
 		return g_steal_pointer (&esp_path);
 	/* if ID key doesn't exist, try ID_LIKE */
 	id_like_id = g_hash_table_lookup (os_release, "ID_LIKE");
+	/* try to fallback to use UEFI removable path if ID_LIKE key doesn't exist */
 	if (id_like_id != NULL) {
 		g_autofree gchar* id_like_path = g_build_filename (base, "EFI", id_like_id, NULL);
 		if (g_file_test (id_like_path, G_FILE_TEST_IS_DIR)) {
 			g_debug ("Using ID_LIKE key from os-release");
 			return g_steal_pointer (&id_like_path);
 		}
+	} else if (fu_device_get_metadata_boolean (device, "FallbacktoRemovablePath")) {
```

Comment:
just to double check - `/etc/os-release` is indeed missing for ubuntu core right?

Reply:
There is a cat from UC18:
```
NAME="Ubuntu Core"
VERSION="18"
ID=ubuntu-core
PRETTY_NAME="Ubuntu Core 18"
VERSION_ID="18"
HOME_URL="https://snapcraft.io/"
BUG_REPORT_URL="http://bugs.launchpad.net/snappy/"
```